### PR TITLE
Do not count context methods as assertion methods (fixes #142)

### LINKED ISSUE #142: max-asserts incorrectly counts called functions on t.context as assertions
### Problem
`ava/max-asserts` incorrectly counts called functions on `t.context` as assertions.

### Example:
```js
/* eslint "ava/max-asserts" : [ "warn", 5 ] */

import test from 'ava';

test.beforeEach( t => {
	t.context.bar = () => 'bar';
});


// this will be fine
test( 'max-asserts ok', t => {
	const foo = 'foo';

	t.true( typeof foo === 'string' );
	t.true( typeof foo === 'string' );
	t.true( typeof foo === 'string' );
	t.true( typeof foo === 'string' );
	t.true( typeof foo === 'string' );
});


// this will count 6 assertions
test( 'max-asserts incorrect count', t => {
	const foo = 'foo';

	// this is counted as an assertion
	const bar1 = t.context.bar();

	// this is not counted as an assertion
	const Bar = t.context.bar;
	const bar2 = Bar();

	t.true( typeof foo === 'string' );
	t.true( typeof foo === 'string' );
	t.true( typeof foo === 'string' );
	t.true( typeof foo === 'string' );
	t.true( typeof foo === 'string' );
});
```

### Expected
No warnings

### Actual
The following warnings are displayed when linting the above file:
```
37:2  warning  Expected at most 5 assertions, but found 6  ava/max-asserts
✖ 1 problem (0 errors, 1 warning)
```

### PR DESCRIPTION
**Pull request by jfmengels**
Do not count context methods as assertion methods (fixes #142)

### CODE REVIEW

File: util.js [hunk 1]
```diff
@@ -52,3 +52,36 @@ exports.getTestModifiers = function getTestModifiers(node) {
...
+	if (node.object.type === 'MemberExpression') {
+		return getMembers(node.object).concat(name);
+	}
+
+	return [name];
+};
+
+exports.getMembers = getMembers;
+
+const assertionMethodsNbArguments = {
```
Comment (sindresorhus) [inline util.js]: I would name it: `assertionMethodNumArgs`. Num is a more common short name for Number.

File: rules/use-t-well.js [hunk 2]
```diff
@@ -3,38 +3,10 @@ const visitIf = require('enhance-visitors').visitIf;
...
-	'deepEqual',
-	'notDeepEqual',
-	'throws',
-	'notThrows',
-	'regex',
-	'notRegex',
-	'ifError',
-	'plan'
-];
+const methods = util.assertionMethods.concat(['end', 'plan']);
```
Comment (sindresorhus) [inline rules/use-t-well.js]: I think this should be in `util.js` too. Name it `util.contextMethods`.

File: util.js (hunk 1 above)
Comment (sindresorhus) [inline util.js]: I think we should use a `Map` here instead.

File: util.js [hunk 3]
```diff
@@ -52,3 +52,36 @@ exports.getTestModifiers = function getTestModifiers(node) {
...
+	pass: 0,
+	regex: 2,
+	notRegex: 2,
+	throws: 1,
+	true: 1,
+	truthy: 1
+};
+
+exports.assertionMethodsNbArguments = assertionMethodsNbArguments;
+exports.assertionMethods = Object.keys(assertionMethodsNbArguments);
```
Comment (sindresorhus) [inline util.js]: And this should be a `Set`.

File: rules/use-t-well.js (hunk 2 above)
Comment (jfmengels) [inline rules/use-t-well.js]: contextMethods? They're not part of `t.context` :confused:

File: util.js (hunk 1 above)
Comment (jfmengels) [inline util.js]: Why the preference for Map and Set? Except for holding objects as keys for Map, I don't know the benefits. Don't mind changing it, as I don't see drawbacks to it, mostly curious.

Comment (jfmengels) [inline util.js]: Changed to Map/Set, but would still like your opinion on its benefits :)

File: rules/use-t-well.js (hunk 2 above)
Comment (jfmengels) [inline rules/use-t-well.js]: Named it executionMethods, as it's called the execution object in the docs.

File: util.js (hunk 1 above)
Comment (sindresorhus) [inline util.js]: They're more proper for data, better API, faster, and can handle any kind of key name.

File: rules/use-t-well.js (hunk 2 above)
Comment (sindresorhus) [inline rules/use-t-well.js]: Ah right. We used to call `t` the test context before we got `t.context`.

**Comment by jfmengels** [comment]
Aw man... forgot `...` wasn't supported in v4... -_-'

**Comment by sindresorhus** [comment]
Looks great! :)

**Pull request merged**